src/controllers/chat.ts

```diff
@@ -0,0 +1,102 @@
+import type { Request, Response, NextFunction } from 'express';
+import type { IChat } from '../models/chat';
+import Chat from '../models/chat';
+import Room from '../models/room';
+import {
+  NOT_FOUND,
+  FORBIDDEN,
+  CREATED,
+  OK,
+  NO_CONTENT,
+} from '../constants/httpStatusCodes';
+import {
+  MESSAGE_NOT_FOUND,
+  MUST_JOIN_ROOM,
+} from '../constants/errorMessages';
+
+const checkChatExists = (chat: IChat, res: Response) => {
+  if (!chat) {
+    return res.status(NOT_FOUND).json({ message: MESSAGE_NOT_FOUND });
+  }
+};
+
+export const createMessage = async(
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    const { message, sender, room } = req.body;
+
+    const roomObj = await Room.findById(room);
```

Comment:
all interaction with the db or data overall should be in service, controller is just a 'middle-man' which receives request from the client, and send it to the service, and then gets response from the service and sends it to the client